### ISSUE TITLE
Textblocks with an empty TextFragment cause problems

### ISSUE THREAD

**Issue by mattberkowitz** [ProseMirror/prosemirror#254]
The easiest way to reproduce this is to create a document using:

```
new ProseMirror({
  doc: defaultSchema.node("doc", {}, defaultSchema.node("paragraph", {}, defaultSchema.text(""))),
  schema: defaultSchema
})
```

When you try to place the cursor in the editor you get the error
/../dist/edit/dompos:291 Uncaught TypeError: Cannot read property 'undefined' of undefined

`coordsAtPosition` goes into this block https://github.com/ProseMirror/prosemirror/blob/master/src/edit/dompos.js#L249-L258 and fails both of the conditionals so `rect` and `side` are undefined

I've also seen a similar issue (I think to do with dragging, but I can't reproduce at the moment) to this where `posAtCoords` is called and `findOffsetInNode` inside that function returns the empty text node and an offset of 1.  Eventually those values get into `textRect` where it fails to construct a range as the offset is greater than the size of the text node (https://github.com/ProseMirror/prosemirror/blob/master/src/edit/dompos.js#L232)

I'm not able to reproduce the empty text node scenario through normals means, only with code. In our app the scenario is being produced by our custom transformer which calls `schema.text(text)` where the `text` variable can be empty, especially on a fresh document

It seems like like the *easiest* solution is for `schema.text("")` to return null rather than a `TextFragment` with a zero-length text node. This also seems consistent with other behavior in ProseMirror (ie when I hit enter to create a new paragraph, it creates a `Paragraph` with a `FlatFragment` that has no child nodes, which is the same thing as `schema.node('paragraph', {}, null)` produces)

I'm happy to make the changes just let me know if you agree or if I should go down the path of getting `coordsAtPos`/`findOffsetInNode` to better handle the empty text node scenario

**Comment by marijnh** [comment]
Empty text fragments are simply forbidden, don't create them. It might be a good idea to have the TextNode constructor raise an error when they are created. Want to submit a patch? (Returning null doesn't seem like a good idea. This is invalid and the code constructing the node should check _before_ trying to create a text node.)

**Comment by mattberkowitz** [comment]
Yep, I'll make it today

**Comment by marijnh** [comment]
Solved via #256
